spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitAutoConfiguration.java

```diff
@@ -186,4 +192,36 @@ public RabbitMessagingTemplate rabbitMessagingTemplate(RabbitTemplate rabbitTemp
 
 	}
 
+	@Configuration(proxyBeanMethods = false)
+	@ConditionalOnClass(RabbitStreamTemplate.class)
+	@ConditionalOnMissingBean(RabbitStreamTemplate.class)
+	@ConditionalOnProperty(prefix = "spring.rabbitmq.listener", name = "type", havingValue = "stream")
+	protected static class RabbitStreamTemplateConfiguration {
+
+		@Bean
+		@ConditionalOnMissingBean(RabbitStreamOperations.class)
+		@ConditionalOnProperty(prefix = "spring.rabbitmq.stream", name = "name")
+		public RabbitStreamTemplate rabbitStreamTemplate(Environment rabbitStreamEnvironment,
+				RabbitProperties properties, ObjectProvider<MessageConverter> messageConverters,
+				ObjectProvider<StreamMessageConverter> streamMessageConverters,
+				ObjectProvider<ProducerCustomizer> producerCustomizers) {
+			RabbitStreamTemplate template = new RabbitStreamTemplate(rabbitStreamEnvironment,
```

Comment:
It sounds like it's calling for a `RabbitStreamTemplateConfigurer` to be consistent with what we're currently doing with `RabbitTemplate`.

Reply:
Agreed.